resharper/resharper-unity/test/data/CSharp/CodeCompletion/Action/RetypeFullNameOnExistingMethod02.cs

```diff
@@ -0,0 +1,10 @@
+// ${COMPLETE_ITEM:OnApplicationPause(bool) { ... \}}
+using UnityEngine;
+
+public class A : MonoBehaviour
```

Comment:
@citizenmatt May be we should use UnterminatedContext reparse in this case? Instead of `var updateMethodDeclaration = TextControlToPsi.GetElement<IMethodDeclaration>(solution, textControl);
`
It inserts __ in caret position and we can work with 
`  public void __{caret}(bool hasFocus)
  {
    // Existing code
  }`
  
  So it will parse as correct method declaration and doesn't break anything